pandas/core/internals/blocks.py

```diff
@@ -2926,7 +2962,7 @@ def _maybe_coerce_values(self, values, dtype=None):
         if dtype is not None:
             if isinstance(dtype, compat.string_types):
                 dtype = DatetimeTZDtype.construct_from_string(dtype)
-            values = values._shallow_copy(tz=dtype.tz)
+            values = type(values)(values, dtype=dtype)
```

Comment:
if we get here, isn't `values` already a DTA?  are we assured that we wont have to do tz_convert?

Reply:
I'm not sure. But if we previously did DatetimeIndex._shallow_copy, we should now have a DatetimeArray I think.